src/crimson/osd/backfill_facades.h

```diff
@@ -35,9 +35,12 @@ struct BackfillState::PeeringFacade {
     return peering_state.get_info().log_tail;
   }
 
-  decltype(auto) get_pg_log() const {
-    return peering_state.get_pg_log();
+  template <class... Args>
+  void scan_log_after(Args&&... args) const {
```

Comment:
It wraps `IndexedLog::scan_log_after()` which is also a template:

```cpp
    template <typename T>
    void scan_log_after(
      const eversion_t &bound, ///< [in] scan entries > bound
      T &&f) const {
        // ...
    }
```

Not sure whether using a variadic template instead of a plain one drastically increases the complexity here. The benefit is emphasizing the fact the method doesn't actually care about any parameter – it just forwards everything.

Anyway, no strong feelings here.